src/servers/src/http/event.rs

```diff
@@ -230,6 +231,92 @@ fn transform_ndjson_array_factory(
         })
 }
 
+#[axum_macros::debug_handler]
+pub async fn test_pipeline(
+    State(log_state): State<LogState>,
+    Query(query_params): Query<LogIngesterQueryParams>,
+    Extension(mut query_ctx): Extension<QueryContext>,
+    TypedHeader(content_type): TypedHeader<ContentType>,
+    payload: String,
+) -> Result<Response> {
+    let handler = log_state.log_handler;
+    let pipeline_name = query_params.pipeline_name.context(InvalidParameterSnafu {
+        reason: "pipeline_name is required",
+    })?;
+
+    let version = to_pipeline_version(query_params.version).context(PipelineSnafu)?;
+
+    let ignore_errors = query_params.ignore_errors.unwrap_or(false);
+
+    let value = extract_pipeline_value_by_content_type(content_type, payload, ignore_errors)?;
+
+    query_ctx.set_channel(Channel::Http);
+    let query_ctx = Arc::new(query_ctx);
+
+    let pipeline = handler
+        .get_pipeline(&pipeline_name, version, query_ctx.clone())
+        .await?;
+
+    let mut intermediate_state = pipeline.init_intermediate_state();
+
+    let mut results = Vec::with_capacity(value.len());
+    for v in value {
+        pipeline
+            .prepare(v, &mut intermediate_state)
+            .map_err(|reason| PipelineTransformSnafu { reason }.build())
+            .context(PipelineSnafu)?;
+        let r = pipeline
+            .exec_mut(&mut intermediate_state)
+            .map_err(|reason| PipelineTransformSnafu { reason }.build())
+            .context(PipelineSnafu)?;
+        results.push(r);
+        pipeline.reset_intermediate_state(&mut intermediate_state);
+    }
+
+    let schema = pipeline.schemas().iter().map(|cs| {
+        let mut map = Map::new();
+        map.insert("name".to_string(), Value::String(cs.column_name.clone()));
+        map.insert(
+            "data_type".to_string(),
+            Value::String(cs.datatype().as_str_name().to_string()),
+        );
+        map.insert(
+            "colume_type".to_string(),
+            Value::String(cs.semantic_type().as_str_name().to_string()),
+        );
+        map.insert(
+            "fulltext".to_string(),
+            Value::Bool(
+                cs.options
+                    .clone()
+                    .is_some_and(|x| x.options.contains_key("fulltext")),
+            ),
+        );
+        Value::Object(map)
+    });
+    let rows = results.into_iter().map(|row| {
+        let row = row
+            .values
+            .into_iter()
+            .map(|v| {
+                v.value_data
+                    .map(|d| {
+                        let mut map = Map::new();
+                        map.insert("value".to_string(), column_data_to_json(d));
```

Comment:
For each row in the result, a more compact format would be preferred
```Json
[
  2436,
  2528,
  "I",
  "ClusterAdapter:enter sendTextDataToCluster\n",
  "INTERACT.MANAGER",
  "2024-05-25 20:16:37.217+0000"
]
```
This is also in consistent with the output format of our http sql api.

Reply:
It's something I've considered. My thought is that this doesn't really match the output of our query. This structure is used here to make it easier to expand on each field result later.

Reply:
Nitpick: If the api is designed for human reading, I would suggest the format to be like 
```JSON
[
  {
    "id1": 2436
  },
  {
    "id2": 2528
  },
  {
    "type": "I"
  },
  {
    "log": "ClusterAdapter:enter sendTextDataToCluster\n"
  },
  {
    "logger": "INTERACT.MANAGER"
  },
  {
    "time": "2024-05-25 20:16:37.217+0000"
  }
]
```

Also, should we limit the number of input lines, like max of 10? In case the input is huge enough to trigger a cpu spike.

Reply:
how about 
```
[
  {
    “key”: “id1”,
    “value": 2436
    “type”: INT
  }
]
```
We also keep schema key. This one is primarily for people, not machines. I think there could be some redundancy.

Reply:
how about 
```
[
  {
    “key”: “id1”,
    “value": 2436
    “type”: INT
  }
]
```
We also keep schema key. This one is primarily for people, not machines. I think there could be some redundancy.